pattern/constraint/thing/ValueConstraint.java

```diff
@@ -195,6 +197,30 @@ public FunctionalIterator<AlphaEquivalence> alphaEquals(ValueConstraint<?> that)
                 .flatMap(a -> a.alphaEqualIf(this.value.equals(that.value)));
     }
 
+    public abstract boolean isSatisfiedByValueOf(ValueConstraint<?> conclusionValueConstraint);
+
+    protected boolean evaluateEqualityPredicate(TypeQLToken.Predicate.Equality predicate, int comparatorOutput) {
```

Comment:
Comparison based 'equality' predicates use this method to evaluate the predicate against the comparator output of the concludable and conclusion value.